modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -145,10 +149,10 @@ protected Message toMessage(Object value, ExecutionContextAdapter operationConte
       if (resultValue.getOutput() instanceof InputStream) {
         ConnectionHandler connectionHandler = (ConnectionHandler) operationContext.getVariable(CONNECTION_PARAM);
         if (connectionHandler != null && supportsStreaming(operationContext.getComponentModel())) {
-          resultValue = resultValue.copy()
-              .output(componentDecoratorFactory
-                  .decorateOutput(new ConnectedInputStreamWrapper((InputStream) resultValue.getOutput(), connectionHandler),
-                                  event.getCorrelationId()))
+          resultValue = resultValue
+              .copy().output(new ConnectedInputStreamWrapper((InputStream) resultValue.getOutput(), connectionHandler,
```

Comment:
This still needs to be decorated by the componentDecoratorFactory

Reply:
right, my bad